test/parallel/test-fs-open.js

```diff
@@ -3,17 +3,12 @@ const common = require('../common');
 const assert = require('assert');
 const fs = require('fs');
 
-let caughtException = false;
+const expected = new RegExp('^Error: ENOENT: no such file or directory, ' +
+                            'open \'/path/to/file/that/does/not/exist\'$');
 
-try {
-  // should throw ENOENT, not EBADF
-  // see https://github.com/joyent/node/pull/1228
+assert.throws(() => {
   fs.openSync('/path/to/file/that/does/not/exist', 'r');
-} catch (e) {
-  assert.strictEqual(e.code, 'ENOENT');
```

Comment:
I see your point @lpinca, I could change this to:

```javascript
assert.throws(() => {
  fs.openSync('/path/to/file/that/does/not/exist', 'r');
}, (err) => {
  assert.strictEqual(err.code, 'ENOENT');
  assert(expected.test(err));
  return true;
});
```

but this is not too common in the test suite, would like to have some opinions from @nodejs/testing